lib/util/addExtraParamsTo.js

```diff
@@ -10,7 +10,11 @@ function addExtraParamsTo(routeArguments) {
     //  is null. WWARNER 08/21/2014
     paramsAndRequest.pop();
 
-    var extraParams = Array.prototype.slice.call(arguments, 1);
```

Comment:
I still don't see how this is any different. You're creating a shallow copy by iterating over it (thats propably what slice does under the hood). I don't see how this will change anything